internal/datastore/mysql/watch.go

```diff
@@ -21,7 +21,12 @@ const (
 //
 // All events following afterRevision will be sent to the caller.
 func (mds *Datastore) Watch(ctx context.Context, afterRevisionRaw datastore.Revision, options datastore.WatchOptions) (<-chan *datastore.RevisionChanges, <-chan error) {
-	updates := make(chan *datastore.RevisionChanges, mds.watchBufferLength)
+	watchBufferLength := options.WatchBufferLength
+	if watchBufferLength <= 0 {
+		watchBufferLength = mds.watchBufferLength
```

Comment:
should probably warn when we do these fallbacks, but perhaps that's a tech debt pass for another time

Reply:
I'll do it as a followup